services/src/main/scala/cromwell/services/metadata/MetadataService.scala

```diff
@@ -129,10 +131,12 @@ object MetadataService {
   def throwableToMetadataEvents(metadataKey: MetadataKey, t: Throwable): List[MetadataEvent] = {
     t match {
       case aggregation: ThrowableAggregation =>
-        aggregation.errorMessages.toList map { message => MetadataEvent(metadataKey.copy(key = s"${metadataKey.key}:message"), MetadataValue(s"${aggregation.exceptionContext}: $message")) }
+        val message = List(MetadataEvent(metadataKey.copy(key = s"${metadataKey.key}[${Random.nextInt(Int.MaxValue)}]:message"), MetadataValue(aggregation.exceptionContext)))
+        val innards = aggregation.throwables.toList flatMap { innard => throwableToMetadataEvents(metadataKey, innard) }
+        message ++ innards
       case other =>
-        val message = List(MetadataEvent(metadataKey.copy(key = s"${metadataKey.key}:message"), MetadataValue(t.getMessage)))
-        val cause = Option(t.getCause) map { cause => throwableToMetadataEvents(metadataKey.copy(key = s"${metadataKey.key}:causedBy"), cause) } getOrElse List.empty
+        val message = List(MetadataEvent(metadataKey.copy(key = s"${metadataKey.key}[${Random.nextInt(Int.MaxValue)}]:message"), MetadataValue(t.getMessage)))
+        val cause = Option(t.getCause) map { cause => throwableToMetadataEvents(metadataKey, cause) } getOrElse List.empty
```

Comment:
Injecting the `Random.nextInt` here _inside_ `throwableToMetadataEvents` on 138 separates the exception from the recursive underlying cause on 139. IMHO that makes debugging much more confusing. These are _not_ separate events, and should be tied together.

Btw, will tech talk later, but as far as I can tell we only have **one** active json schema for exceptions, the one mentioned in the [pseudo-CFG](https://github.com/broadinstitute/cromwell/issues/2037#issuecomment-282800499).

The inactive json schema that may have triggered this whole issue appears to be an artifact of the [migration](https://github.com/broadinstitute/cromwell/blob/509ff2dc2bc2208546a76307791af05816ec83f5/database/migration/src/main/scala/cromwell/database/migration/metadata/table/FailureEventTableMigration.scala#L23) [code](https://github.com/broadinstitute/cromwell/blob/509ff2dc2bc2208546a76307791af05816ec83f5/database/migration/src/main/scala/cromwell/database/migration/metadata/table/FailureEventTableMigration.scala#L47) not matching the active code, deleted in red above.

I'm fine upgrading and migrating our current the json [schema](http://json-schema.org/) to something non-recursive, but would propose something along the lines of:

```json
{
  "$ref": "#/definitions/failures",
  "definitions": {
    "failures": {
      "type": "array",
      "items": {
        "$ref": "#/definitions/failure"
      }
    },
    "failure": {
      "required": ["message"],
      "properties": {
        "message": {
          "type": "string"
        },
        "causedBy": {
          "type": "array",
          "items": {
            "type": "string"
          }
        }
      }
    }
  }
}
```

This schema would change the recursive:

```json
{
  "failures": [
    {
      "message": "Failed to upload authentication file",
      "causedBy": {
        "message": "Error getting access token for service account: ",
        "causedBy": {
          "message": "connect timed out"
        }
      }
    }
  ]
}
```

to the more condensed:

```json
{
  "failures": [
    {
      "message": "Failed to upload authentication file",
      "causedBy": [
        "Error getting access token for service account: ",
        "connect timed out"
      ]
    }
  ]
}
```

and would still allow one to figure out what exceptions are connected, or not:

```json
{
  "failures": [
    {
      "message": "Internal unconnected exception with no cause by"
    },
    {
      "message": "A separate I/O exception that failed to upload authentication file",
      "causedBy": [
        "Error getting access token for service account: ",
        "connect timed out"
      ]
    }
  ]
}
```

Also fine with re-migrating the old data to bring it up to a current json schema implementation.